app/services/api/registryService.js

```diff
@@ -35,6 +35,63 @@ angular.module('portainer.services')
     return deferred.promise;
   };
 
+  service.catalog = function(id) {
```

Comment:
I'd prefer to have a model created for each entity like we do for other Docker resources (check `app/models/docker`, I think `app/models/registry` would be fine).

Reply:
I think this is needed for docker cause we transform data between docker api and portainer model
Here angular use raw data from registry api
WDYT ?

Reply:
I think that this is more of a pattern we use in the Portainer codebase, we could have gone with raw data from the Docker API as well.

But to be consistent, I'd rather use a Portainer model for the registry data as well.

Reply:
OK will do that soon